dali/core/fast_div_test.cu

```diff
@@ -0,0 +1,230 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/core/fast_div.h"  // NOLINT
+#include <gtest/gtest.h>
+#include <random>
+#include <iostream>
+#include "dali/test/device_test.h"
+#include "dali/core/dev_buffer.h"
+#include "dali/core/cuda_event.h"
+
+namespace dali {
+
+template <uint32_t divisor>
+void TestDiv32() {
+  fast_div<uint32_t> fast = divisor;
+  std::cerr << "Testing x / " << divisor << std::endl;
+  for (uint32_t x = 0; x < 0xffffffffu; x++) {
+    ASSERT_EQ(x / divisor, x / fast) << " when dividing " << x << " / " << divisor;
+  }
+}
+
+// This test is disabled because it's ridiculously slow - only run when touching fast_div
+TEST(FastDiv, DISABLED_U32_Host) {
```

Comment:
Can you add a fast variant for only a selected `x` values?

Reply:
Maybe... I'd have to select them carefully, though.